cloud/cluster/node_update.go

```diff
@@ -0,0 +1,39 @@
+package cluster
+
+import (
+	"fmt"
+)
+
+// NodeUpdate represents a change to the cluster
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+// Helper functions to create NodeUpdates
+
+func NewAdd(node Node) NodeUpdate {
+	return NodeUpdate{
+		NodeAdded,
+		node.Id(),
+		node,
+	}
+}
+
+func NewRemove(id NodeId) NodeUpdate {
+	return NodeUpdate{
+		UpdateType: NodeRemoved,
+		Id:         id,
+	}
+}
+
+type NodeUpdateSorter []NodeUpdate
```

Comment:
The cluster test and state test both use node update sort to compare expected vs actual. How do you feel about sorting updates only in the test?